### PR TITLE
Add simple form base app in security-authentication-mechanisms guide

### LINKED ISSUE #37645: Docs: security-authentication-mechanisms Guide
### Describe the bug

Going through the [security-authentication-mechanisms](https://quarkus.io/version/main/guides/security-authentication-mechanisms) guide I found some issues.

[Form-based authentication](https://quarkus.io/guides/security-authentication-mechanisms#form-auth) I personally thing this section can be improved. For me it wasn't perfectly clear if the client side verification by javascript was executed automatically when send request or it should be some kid of function/method which will connected to some kind of form. It should not clear from that example. Maybe point to some resource where it's possible to get more info about form base authentication. 
There would be nice to show like "traditional" way with form in html. 

Just small thing the link to [Duration#parse() javadoc](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/time/Duration.html#parse(java.lang.CharSequence)) should probably point to java 17 as Quarkus move to java 17.

Rest of the guide seems good and don't see any problem as more info about other mechanisms have own guide. Also I know that form base authentication is not primary type of authentication but for some smaller/entry project this can be go to type of authentication.

### Expected behavior

_No response_

### Actual behavior

_No response_

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### Quarkus version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by jedla97**
Fixes https://github.com/quarkusio/quarkus/issues/37645

Adding simple app example for form-base security which I was missing when reading the guide

### CODE REVIEW

File: docs/src/main/asciidoc/security-authentication-mechanisms.adoc [hunk 1]
```diff
@@ -81,6 +81,69 @@ The resulting digest is used as a key for AES-256 encryption of the cookie value
...
+public class FormResource {
+
+    @GET
+    @Path("hello")
+    @RolesAllowed("user")
+    public String hello(@Context SecurityContext security) {
+        return "Hello " + security.getUserPrincipal().getName();
+    }
+
+    @GET
```
Comment (sberyozkin) [inline docs/src/main/asciidoc/security-authentication-mechanisms.adoc:100]: IMHO we can just drop this example, for the user facing application one would not return Strings but for example here produce HTML with Qute. This specific doc's goal is to give an overview of the authentication mechanisms only. 
Perhaps it is time to introduce a dedicated doc related to the form authentication

File: docs/src/main/asciidoc/security-authentication-mechanisms.adoc [hunk 2]
```diff
@@ -81,6 +81,69 @@ The resulting digest is used as a key for AES-256 encryption of the cookie value
...
+# Define testing user
+quarkus.security.users.embedded.enabled=true
+quarkus.security.users.embedded.plain-text=true
+quarkus.security.users.embedded.users.alice=alice 
+quarkus.security.users.embedded.roles.alice=user
+----
+
+[IMPORTANT]
+====
+Configuring user names, secrets, and roles in the application.properties file is appropriate only for testing scenarios. For securing a production application, it is crucial to use a database to store this information.
```
Comment (sberyozkin) [inline docs/src/main/asciidoc/security-authentication-mechanisms.adoc:125]: LDAP is also an option, it also uses DB indirectly, but may be it is worth saying `database or LDAP` and link to the basic authentication with JPA doc for an example.

File: docs/src/main/asciidoc/security-authentication-mechanisms.adoc [hunk 3]
```diff
@@ -81,6 +81,69 @@ The resulting digest is used as a key for AES-256 encryption of the cookie value
...
+    <input type="text" placeholder="Username" name="j_username" required>
+    <label>Password</label>
+    <input type="password" placeholder="Password" name="j_password" required>
+    <button type="submit">Login</button>
+</form>
+----
+
+[NOTE]
+====
+Endpoind for verification `j_security_check` and name of parameters for username (`j_username`) and password (`j_password`) can be changed by setting it's configuration properties.
```
Comment (sberyozkin) [inline docs/src/main/asciidoc/security-authentication-mechanisms.adoc:143]: A few typos, but IMHO this note is unnecessary, the configuration section follows just below and it is clear all of these properties can be changed

**Review by sberyozkin — COMMENTED** [review]
Thanks @jedla97, LGTM, I'm only proposing to try to minimize the new text as this overview of the authentication mechanisms doc should only give some basic overview of a specific mechanism as opposed to for example being a tutorial on the Form authentication.

IMHO it will make sense to create a dedicated Form authentication doc later with a Form auth tutorial, and a quickstart, as it looks like Form authentication is starting to dominate in this doc

File: docs/src/main/asciidoc/security-authentication-mechanisms.adoc [hunk 4]
```diff
@@ -81,6 +81,69 @@ The resulting digest is used as a key for AES-256 encryption of the cookie value
...
+
+[source,java]
+----
+@Path("")
+public class FormResource {
+
+    @GET
+    @Path("hello")
+    @RolesAllowed("user")
+    public String hello(@Context SecurityContext security) {
```
[the remark this answers is not in the document]
Comment (jedla97) [reply docs/src/main/asciidoc/security-authentication-mechanisms.adoc:96]: Theoretically in this example is not necessary use `SecurityContext/SecurityIdentity` but just return simple string like `Successfully logged in`.

Or I can create that small section but where should I put it?

File: docs/src/main/asciidoc/security-authentication-mechanisms.adoc (hunk 1 above)
Comment (jedla97) [reply docs/src/main/asciidoc/security-authentication-mechanisms.adoc:100]: Ok I'll remove it

**Comment by jedla97** [comment]
@sberyozkin I tried to keep it at minimum. I'll update all the comments just one have question.

> IMHO it will make sense to create a dedicated Form authentication doc later with a Form auth tutorial, and a quickstart, as it looks like Form authentication is starting to dominate in this doc

It would be nice to have it separate, for me it was that I just missing where to start with form. As I believe that lots of small projects or small learning projects don't need complex security as using oidc for example.

**Comment by jedla97** [comment]
> Does it sound reasonable ?

@sberyozkin Yes it does!

> So lets add a few more clarifications with your PR (for ex, how to configure) but then, in a follow up, create a security-form-authentication.adoc and move most of the current Form related content there and only retain a short overview in this doc, with a link to the new one.

Just for clarification, who should create the follow-up? I can work on this a bit, but I'm not sure I'll cover all the technical details correctly. If it should be similar to [this](https://quarkus.io/version/main/guides/security-getting-started-tutorial) for example I can do it.

**Review by sberyozkin — APPROVED** [review]
Thanks @jedla97

**Pull request merged**